internal/backends/sqlite/metadata/registry.go

```diff
@@ -147,22 +147,25 @@ func (r *Registry) CollectionCreate(ctx context.Context, dbName string, collecti
 
 	tableName := r.CollectionToTable(collectionName)
 
-	// TODO use transactions
-	// https://github.com/FerretDB/FerretDB/issues/2747
-
-	query := fmt.Sprintf("CREATE TABLE %q (_ferretdb_sjson TEXT)", tableName)
-	if _, err = db.ExecContext(ctx, query); err != nil {
-		var e *sqlite.Error
-		if errors.As(err, &e) && e.Code() == sqlitelib.SQLITE_ERROR {
-			return false, nil
+	err = pool.InTransaction(ctx, db, func(tx *sql.Tx) error {
+		query := fmt.Sprintf("CREATE TABLE %q (_ferretdb_sjson TEXT)", tableName)
+		if _, err = tx.ExecContext(ctx, query); err != nil {
+			var e *sqlite.Error
+			if errors.As(err, &e) && e.Code() == sqlitelib.SQLITE_ERROR {
```

Comment:
I wonder if in this case we have the same problem as with pg: if an error happened during the transaction, the next query won't be executed. Is it possible to test/reproduce this to ensure that `insert` will be called, and not ignored?

Reply:
What is that problem?

Reply:
Because if the second query in the transaction is not executed, the result will be different than what we want.

Reply:
Why do we need to execute the second query in this case? we return `false,nil` if there's the `SQLITE_ERROR` intentionally.